src/cs/Ssh/SecureStream.cs

```diff
@@ -206,6 +234,33 @@ public async Task ConnectAsync(CancellationToken cancellation = default)
 		this.ConnectCompletion.TrySetResult(true);
 	}
 
+	/// <summary>
+	/// Re-initiates the SSH session over a NEW transport stream by exchanging initial messages
+	/// with the remote server. Waits for the secure reconnect handshake to complete. Additional
+	/// message processing is kicked off as a background task chain.
+	/// </summary>
+	/// <exception cref="SshConnectionException">The connection failed due to a protocol
+	/// error.</exception>
+	/// <exception cref="TimeoutException">The ConnectTimeout property is set and the initial
+	/// version exchange could not be completed within the timeout.</exception>
+	/// <remarks>
+	/// Applies only to a secure stream client. (The secure stream server handles reconnections
+	/// automatically during the session handshake.)
+	/// </remarks>
+	public async Task ReconnectAsync(
+		Stream transportStream,
+		CancellationToken cancellation = default)
+	{
+		if (!(this.Session is SshClientSession clientSession))
+		{
+			throw new InvalidOperationException("Cannot reconnect SecureStream server.");
+		}
+
+		this.TransportStream = transportStream ??
```

Comment:
Does the previous transport stream need to be closed?

Reply:
Or should we check if the session was connected and throw if not (no reconnecting of a session that haven't connected yet)?

Reply:
> Does the previous transport stream need to be closed?

In a disconnected state, the previous stream would already be closed.

> should we check if the session was connected and throw if not

`clientSession.ReconnectAsync()` on line 261 does that.